crates/bitcoin-da/src/service.rs

```diff
@@ -276,22 +276,31 @@ impl BitcoinService {
         });
     }
 
+    /// Retrieves the most recent spendable UTXO from the transaction chain on startup.
+    /// `get_prev_utxo` is called on service startup in `spawn_da_queue`.
+    /// On very first startup, return any most recent UTXO.
+    /// Any subsequent startup would return latest reveal TX first output.
+    /// Sorts by (confirmations - ancestor_count), where lower values indicate more recent transactions.
     #[instrument(level = "trace", skip_all, ret)]
     async fn get_prev_utxo(&self) -> Result<Option<UTXO>, anyhow::Error> {
         let mut previous_utxos = self
             .client
             .list_unspent(Some(0), None, None, Some(true), None)
             .await?;
 
-        previous_utxos.retain(|u| u.spendable && u.solvable);
+        // Make sure to retain only utxos from reveal TXs by matching on REVEAL_OUTPUT_AMOUNT
+        previous_utxos.retain(|u| {
+            u.spendable && u.solvable && u.amount == Amount::from_sat(REVEAL_OUTPUT_AMOUNT)
```

Comment:
u.amount >= DUST_VALUE

Reply:
since reveal is using commit vout0, and commit vout1 is free

wouldn't that mean in the sort below, we'll  choose the reveal output?

my concern is what happens when we don't find an output to use

was that taken care of where this func is called? cant remember

Reply:
yeah that is fine